src/main/java/fi/otavanopisto/kuntaapi/server/integrations/vcard/VCardTranslator.java

```diff
@@ -0,0 +1,271 @@
+    if (vCardNumbers != null && !vCardNumbers.isEmpty()) {
+      List<ContactPhone> result = new ArrayList<>(vCardNumbers.size());
+      for (Telephone vCardNumber : vCardNumbers) {
+        ContactPhone contactPhone = new ContactPhone();
+        if (StringUtils.isNotBlank(vCardNumber.getText())) {
+          contactPhone.setNumber(vCardNumber.getText());
+          contactPhone.setType(getPhoneType(vCardNumber.getTypes()));
+          result.add(contactPhone);
+        }
+      }
+    }
+    
+    return Collections.emptyList();
+  }
+
+  private List<Address> translateAddresses(List<ezvcard.property.Address> vCardAddresses) {
+    if (vCardAddresses != null && !vCardAddresses.isEmpty()) {
+      List<Address> result = new ArrayList<>(vCardAddresses.size());
+      
+      for (ezvcard.property.Address vCardAddress : vCardAddresses) {
+        Address address = translateAddress(vCardAddress);
+        if (address != null) {
+          result.add(address);
+        }
+      }
+      
+      return result;
+    }
+    
+    return Collections.emptyList();
+  }
+
+  private Address translateAddress(ezvcard.property.Address vCardAddress) {
+    if (vCardAddress == null) {
+      return null;
+    }
+    
+    Address result = new Address();
+    
+    result.setType(getAddressType(vCardAddress.getTypes()));
+    result.setPostOfficeBox(getString(vCardAddress.getPoBox()));
+    result.setPostalCode(getString(vCardAddress.getPostalCode()));
+    result.setPostOffice(getString(vCardAddress.getRegion()));
+    result.setStreetAddress(getLocalizedString(vCardAddress.getStreetAddress()));
+    result.setMunicipality(null);
+    result.setCountry(getString(vCardAddress.getCountry(), VCardConsts.DEFAULT_COUNTRY));
+    result.setQualifier(null);
+
+    return result;
+  }
+
+  private String getAddressType(List<AddressType> types) {
+    if (types != null && !types.isEmpty()) {
+      return types.get(0).getValue();
+    }
+
+    return null;
+  }
+
+  private String getPhoneType(List<TelephoneType> types) {
+    if (types != null && !types.isEmpty()) {
+      return types.get(0).getValue();
+    }
+
+    return null;
+  }
+
+  private List<LocalizedValue> getLocalizedString(String value) {
+    if (StringUtils.isNotBlank(value)) {
+      List<LocalizedValue> result = new ArrayList<>(1);
+      
+      LocalizedValue localizedValue = new LocalizedValue();
+      localizedValue.setLanguage(VCardConsts.DEFAULT_LANGUAGE);
+      localizedValue.setValue(value);
+      
+      result.add(localizedValue);
+      
+      return result;
+    }
+
+    return Collections.emptyList();
+  }
+
+  private String getOrganization(Organization organization) {
+    if (organization != null && organization.getValues() != null && !organization.getValues().isEmpty()) {
+      return organization.getValues().get(0);
+    }
+    
+    return null;
+  }
+  
+  private List<String> getOrganizationUnits(List<Organization> organizations) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getOrganizationUnits" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)